### ISSUE TITLE
[windows desktop]Korean IME In hanja(한자) key problem

### ISSUE THREAD

**Issue by kabyeon** [flutter/flutter#74819]
input Chinese characters, conversion is not possible.

my input sequence.
ㄱ ,ㅏ , hanja(한자) key , 1 

expected result
家

actual result
가

![image](https://user-images.githubusercontent.com/1699070/106030768-d4f38700-6111-11eb-8a82-1a6cf13f2e68.png)

![image](https://user-images.githubusercontent.com/1699070/106032074-5ac40200-6113-11eb-802d-59657d4c8292.png)

![image](https://user-images.githubusercontent.com/1699070/106031018-16843200-6112-11eb-9c24-98924c0d647b.png)

<details>
<summary>flutter doctor -v</summary>

```bash
C:\Users\kabyeon\dev\dart\textinput>flutter doctor -v
[√] Flutter (Channel master, 1.26.0-13.0.pre.194, on Microsoft Windows [Version 10.0.19042.746], locale ko-KR)
    • Flutter version 1.26.0-13.0.pre.194 at C:\Users\kabyeon\dev\dart\flutter
    • Framework revision 71aec53acb (11 hours ago), 2021-01-26 21:53:04 -0800
    • Engine revision b11bef83a5
    • Dart version 2.12.0 (build 2.12.0-263.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\kabyeon\AppData\Local\Android\sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.8.2)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.8.30717.126
    • Windows 10 SDK version 10.0.18362.0

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.52.1)
    • VS Code at C:\Users\kabyeon\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.18.1

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19042.746]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 88.0.4324.104
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 88.0.705.50

• No issues found!
```
</details>

**Comment by TahaTesser** [comment]
Hi @kabyeon 
Thanks for filing the issue, I can reproduce using Korean input on Windows
![image](https://user-images.githubusercontent.com/48603081/106134461-918a2e80-618c-11eb-9add-4955db259571.png)

<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: Center(
        child: Form(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              TextField(
                decoration: InputDecoration(
                  hintText: "Hint ",
                ),
              ),
              TextField(
                decoration: InputDecoration(
                  hintText: "Hint ",
                ),
              ),
              TextField(
                decoration: InputDecoration(
                  hintText: "Hint ",
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

``` 

</details>

<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel master, 1.26.0-13.0.pre.194, on Microsoft Windows [Version 10.0.19042.746], locale ko-KR)
    • Flutter version 1.26.0-13.0.pre.194 at C:\Users\Taha\Code\flutter_master
    • Framework revision 71aec53acb (30 hours ago), 2021-01-26 21:53:04 -0800
    • Engine revision b11bef83a5
    • Dart version 2.12.0 (build 2.12.0-263.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\Taha\Code\SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = C:\Users\Taha\Code\SDK
    • Java binary at: C:\Users\Taha\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.8.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.8.30804.86
    • Windows 10 SDK version 10.0.18362.0

[✓] Android Studio (version 4.1.0)
    • Android Studio at C:\Users\Taha\Code\android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code (version 1.52.1)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.18.1

[✓] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19042.746]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 87.0.4280.141
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 88.0.705.50

• No issues found!
```
=

</details>

Thank you

**Comment by cbracken** [comment]
@kabyeon thanks for filing this. Taking a look.

For others working with a non-Korean keyboard, the right-control key on my keyboard functions as 한자 키 (Hanja conversion key).

**Comment by cbracken** [comment]
Entering 가 then pressing the Hanja key results in a crash today:
```
           ══╡ EXCEPTION CAUGHT BY SERVICES LIBRARY ╞══════════════════════════════════════════════════════════
           The following assertion was thrown during a platform message callback:
           Attempted to send a key down event when no keys are in keysPressed. This state can occur if the key
           event being sent doesn't properly set its modifier flags. This was the event:
           RawKeyDownEvent#9221f(logicalKey: LogicalKeyboardKey#000e5(keyId: "0x107000000e5", keyLabel: "",
           debugName: "Unknown Windows key code 229"), physicalKey: PhysicalKeyboardKey#700e0(usbHidUsage:
           "0x000700e0", debugName: "Control Left")) and its data: Instance of 'RawKeyEventDataWindows'
           'package:flutter/src/services/raw_keyboard.dart':
           Failed assertion: line 638 pos 12: 'event is! RawKeyDownEvent || _keysPressed.isNotEmpty'

           Either the assertion indicates an error in the framework itself, or we should provide substantially
           more information in this error message to help you determine and fix the underlying cause.
           In either case, please report this assertion by filing a bug on GitHub:
             https://github.com/flutter/flutter/issues/new?template=2_bug.md

           When the exception was thrown, this was the stack:
           #2      RawKeyboard._handleKeyEvent (package:flutter/src/services/raw_keyboard.dart:638:12)
           #3      BasicMessageChannel.setMessageHandler.<anonymous closure>
           (package:flutter/src/services/platform_channel.dart:73:49)
           #4      BasicMessageChannel.setMessageHandler.<anonymous closure>
           (package:flutter/src/services/platform_channel.dart:72:47)
           #5      _DefaultBinaryMessenger.handlePlatformMessage (package:flutter/src/services/binding.dart:284:33)
           #6      _invoke3.<anonymous closure> (dart:ui/hooks.dart:227:15)
           #10     _invoke3 (dart:ui/hooks.dart:226:10)
           #11     PlatformDispatcher._dispatchPlatformMessage (dart:ui/platform_dispatcher.dart:520:7)
           #12     _dispatchPlatformMessage (dart:ui/hooks.dart:90:31)
           (elided 5 frames from class _AssertionError and dart:async)
           ════════════════════════════════════════════════════════════════════════════════════════════════════
```

/cc @dkwingsmt for thoughts before I dig into this further.

**Comment by cbracken** [comment]
Verified that the crash is related to #73967, which was previously handled in the embedder but removed in https://github.com/flutter/engine/pull/23465/files#diff-a85c8141ed41c80633ead050389bc33164a928f1b3c284cada1beec4d09af41aL386-L391. 

Discussed briefly offline about handling keycode 229. The plan is to do this in the framework, but that hasn't landed yet. Either I will land a partial revert of just the patch in question above or @dkwingsmt will land something in the framework quickly.

**Comment by cbracken** [comment]
Re-opening since #77039 only fixed the crash, not Hanja support.

**Comment by Jungeol** [comment]
same problem
<details><summary>flutter doctor -v</summary>
<p>
[✓] Flutter (Channel stable, 3.7.12, on Microsoft Windows [Version 10.0.19044.2846], locale ko-KR)
    • Flutter version 3.7.12 on channel stable at C:\Users\jgsim\lib\flutter_windows_3.7.3-stable\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4d9e56e694 (3 weeks ago), 2023-04-17 21:47:46 -0400
    • Engine revision 1a65d409c7
    • Dart version 2.19.6
    • DevTools version 2.20.1

[✗] Windows Version (Unable to confirm if installed Windows version is 10 or greater)

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at C:\Users\jgsim\AppData\Local\Android\sdk
    • Platform android-33-ext4, build-tools 33.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-9505619)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2022 17.5.4)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.5.33530.505
    • Windows 10 SDK version 10.0.22000.0

[✓] Android Studio (version 2022.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-9505619)

[✓] VS Code (version 1.77.3)
    • VS Code at C:\Users\jgsim\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19044.2846]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 113.0.5672.63
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 113.0.1774.35

[✓] HTTP Host Availability
    • All required HTTP hosts are available
</p>
</details> 
<details><summary>windows specs</summary>
<p>

edition:	Windows 10 Pro
version:	21H2
installation date:	‎2021-‎07-‎12
OS build	19044.2846
experience	Windows Feature Experience Pack 120.2212.4190.0

</p>
</details>

**Comment by huycozy** [comment]
This issue seems to have been fixed on the latest stable channel `3.19.2` as I don't see it there:

![image](https://github.com/flutter/flutter/assets/104349824/8ac1ac8d-1859-42d6-b68f-eefd63158797)

<details>
<summary>flutter doctor -v (stable)</summary>

```bash
[!] Flutter (Channel stable, 3.19.2, on Microsoft Windows [Version 10.0.22631.2861], locale en-US)
    • Flutter version 3.19.2 on channel stable at D:\Working\GitHub\flutter
    ! Warning: `flutter` on your path resolves to D:\Working\GitHub\Aliases\flutter.bat, which is not inside your
      current Flutter SDK checkout at D:\Working\GitHub\flutter. Consider adding D:\Working\GitHub\flutter\bin to the
      front of your path.
    ! The dart binary is not on your path. Consider adding D:\Working\GitHub\flutter\bin to your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7482962148 (2 days ago), 2024-02-27 16:51:22 -0500
    • Engine revision 04817c99c9
    • Dart version 3.3.0
    • DevTools version 2.31.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly
      to perform update checks and upgrades.

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at D:\android-sdk-windows
    • Platform android-30, build-tools 33.0.2
    • Java binary at: C:\Program Files\Common Files\Oracle\Java\javapath\java.exe
    • Java version Java(TM) SE Runtime Environment (build 20.0.1+9-29)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.9.2)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.9.34622.214
    • Windows 10 SDK version 10.0.22621.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).

[√] VS Code (version 1.87.0)
    • VS Code at C:\Users\Huy\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.82.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22631.2861]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 122.0.6261.95
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 100.0.1185.36

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.
```
</details>



I'll close this issue as fixed on the latest channel. If you still experience this issue, kindly leave a comment below and I will reopen this. Thank you!
